test/functional/framework.go

```diff
@@ -154,6 +154,24 @@ func (f *FluentdFunctionalFramework) Deploy() (err error) {
 	if err = f.test.Client.Get(f.pod); err != nil {
 		return err
 	}
+	log.V(2).Info("waiting for service endpoints to be ready")
+	serviceUp := false
+	count := 0
+	for count < 10 && !serviceUp {
```

Comment:
Wouldnt this be better resolved with the Poll functions we've used elsewhere

Reply:
updated to use `wait.PollImmediate`